pkg/apimachinery/registered/registered.go

```diff
@@ -59,15 +69,27 @@ func init() {
 	}
 }
 
-// RegisterVersions add the versions the registeredVersions.
-func RegisterVersions(versions ...unversioned.GroupVersion) {
-	for _, v := range versions {
+// RegisterVersions adds the given group versions to the list of registered group versions.
+func RegisterVersions(availableVersions []unversioned.GroupVersion) {
+	for _, v := range availableVersions {
 		registeredVersions[v] = struct{}{}
 	}
 }
 
-// EnableVersions add the versions to the enabledVersions. The caller of this
-// function is responsible to add the version to 'latest' and 'Scheme'.
+// RegisterGroup adds the given group to the list of registered groups.
+func RegisterGroup(groupMeta apimachinery.GroupMeta) error {
```

Comment:
How about moving all the functions that are moved from latest.go to their own file, say "group_meta.go"? The existing code in this file is not focused on the concept of group, while these functions are, so we can naturally split them in two files.